test-shunting-yard.cpp

```diff
@@ -1173,3 +1173,11 @@ TEST_CASE("Exception management") {
   REQUIRE_THROWS(ecalc2.compile("map()['hello']]"));
   REQUIRE_THROWS(ecalc2.compile("map(['hello']]"));
 }
+
+TEST_CASE("Get variables") {
+  calculator c("a + sin(b) - c**2 / d");
+  auto tmp = c.get_variables();
+  auto vars = std::set<std::string>(tmp.begin(), tmp.end());
+  auto expected = std::set<std::string>{"a", "b", "c", "d"};
+  REQUIRE(vars == expected);
+}
```

Comment:
Using inline expressions is useful in a few ways, one of them is that it saves you the trouble of coming up with names for variables that will only be used once. In this case, the name `tmp` is not informative so we might as well not give it a name, e.g.:

```suggestion
TEST_CASE("Get variables") {
  calculator c("a + sin(b) - c**2 / d");
  auto expectedVars = std::unordered_set<std::string>{"a", "b", "c", "d"};
  REQUIRE(c.get_variables() == expectedVars);
}
```